File: FurnitureStore/FurnitureStore.Models/Product/Product.cs
using System.ComponentModel.DataAnnotations;

namespace FurnitureStore.Models.Product
{
    public class Product
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        [Required(AllowEmptyStrings = false, ErrorMessage = "Barcode is required")]
        [MinLength(6)]
        [MaxLength(15)]
        public string Barcode { get; set; }
        public double Price { get; set; }
        public string Dimensions { get; set; }
        public bool IsAvailableInStore { get; set; }
        public bool IsAvailableOnline { get; set; }
        public long SubcategoryId { get; set; }
        public Subcategory.Subcategory Subcategory { get; set; }
        public virtual ICollection<ProductPicture.ProductPicture> ProductPictures { get; set; } = new List<ProductPicture.ProductPicture>();

    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/product/product.dart
import 'package:furniturestore_admin/models/product_pictures/product_pictures.dart';
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class ProductModel {
  int? id;
  String? name;
  String? description;
  String? barcode;
  double? price;
  String? dimensions;
  bool? isAvailableInStore;
  bool? isAvailableOnline;
  int? subcategoryId;
  List<ProductPicturesModel>? pictures;

  ProductModel(
      {this.id,
      this.name,
      this.description,
      this.barcode,
      this.price,
      this.dimensions,
      this.isAvailableInStore,
      this.isAvailableOnline,
      this.subcategoryId,
      this.pictures});

  factory ProductModel.fromJson(Map<String, dynamic> json) =>
      _$ProductModelFromJson(json);

  Map<String, dynamic> toJson() => _$ProductModelToJson(this);
}
